rotkehlchen/chain/avalanche/manager.py

```diff
@@ -0,0 +1,229 @@
+        tx_receipt = self.covalent.get_transaction_receipt(tx_hash)
+        if tx_receipt is False:
+            tx_receipt = self.w3.eth.get_transaction(tx_hash).__dict__  # type: ignore
+            return tx_receipt
+        try:
+            # Turn hex numbers to int
+            tx_receipt.pop('from_address_label', None)
+            tx_receipt.pop('to_address_label', None)
+            block_number = tx_receipt['block_height']
+            tx_receipt['blockNumber'] = tx_receipt.pop('block_height', None)
+            tx_receipt['cumulativeGasUsed'] = tx_receipt.pop('gas_spent', None)
+            tx_receipt['gasUsed'] = tx_receipt['cumulativeGasUsed']
+            successful = tx_receipt.pop('successful', None)
+            tx_receipt['status'] = 1 if successful else 0
+            tx_receipt['transactionIndex'] = 0
+            # TODO input and nonce is decoded in Covalent api, encoded in future
+            tx_receipt['input'] = '0x'
+            tx_receipt['nonce'] = 0
+            for index, receipt_log in enumerate(tx_receipt['log_events']):
+                receipt_log['blockNumber'] = block_number
+                receipt_log['logIndex'] = receipt_log.pop('log_offset', None)
+                receipt_log['transactionIndex'] = 0
+                tx_receipt['log_events'][index] = receipt_log
+        except (DeserializationError, ValueError) as e:
+            raise RemoteError(
+                f'Couldnt deserialize transaction receipt data from covalent {tx_receipt}',
+            ) from e
+        return tx_receipt
+
+    def call_contract(
+            self,
+            contract_address: ChecksumEthAddress,
+            abi: List,
+            method_name: str,
+            arguments: Optional[List[Any]] = None,
+    ) -> Any:
+        """Performs an eth_call to an ethereum contract
+
+        May raise:
+        - RemoteError if Covalent is used and there is a problem with
+        reaching it or with the returned result
+        - BlockchainQueryError if web3 is used and there is a VM execution error
+        """
+
+        contract = self.w3.eth.contract(address=contract_address, abi=abi)
+        try:
+            method = getattr(contract.caller, method_name)
+            result = method(*arguments if arguments else [])
+        except (ValueError, BadFunctionCallOutput) as e:
+            raise BlockchainQueryError(
+                f'Error doing call on contract {contract_address}: {str(e)}',
+            ) from e
+        return result
+
+    def get_logs(
+            self,
+            contract_address: ChecksumEthAddress,
+            event_name: str,
+            from_timestamp: Timestamp,
+            to_timestamp: Timestamp,
+    ) -> Any:
+        result = self.covalent.get_transactions(
+            contract_address,
+            from_timestamp,
+            to_timestamp,
+            True,
+        )
+        events = []
+        for transaction in result:
+            if transaction['log_events']:
```

Comment:
All these Dict accesses can raise `KeyError` and the strptime can raise `ValueError`. These should be somehow handled